GitHub actions

chaws opened the pull request:
This PR shows how Github actions CI ensures code quality by running tests, lint and format-checking.

jscook2345 commented:
Looks really good!

One suggestion:

Use the `gitlabci-lint` action instead. See the example here:

https://github.com/sirupsen/logrus/blob/master/.github/workflows/ci.yaml

jscook2345 commented:
Once you merge this I will help add some more linters that are disabled by default but very useful as well!

chaws commented:
Thanks @jscook2345 ! I've incorporated your suggestions and also split the thing into 3 jobs: lint, format and coverage (which runs tests).

I'm merging this now so that next PR's will run through the checks. Will be waiting for more linters :)
